src/Web/Endpoints/Sheets.cs

```diff
@@ -0,0 +1,45 @@
+﻿
+using DnDCharacterSheet.Application;
+using DnDCharacterSheet.Application.Common.Models;
+using DnDCharacterSheet.Application.Sheets.Commands.CreateSheet;
+using DnDCharacterSheet.Application.Sheets.Commands.DeleteSheet;
+using DnDCharacterSheet.Application.Sheets.Queries.GetSheets;
+
+namespace DnDCharacterSheet.Web.Endpoints;
+
+public class Sheets : EndpointGroupBase
+{
+    public override void Map(WebApplication app)
+    {
+        app.MapGroup(this)
+            .RequireAuthorization()
+            .MapGet(GetSheets)
+            .MapGet(GetUserSheets, "user")
+            .MapGet(GetSheet, "{id}")
+            .MapPost(CreateSheet)
+            .MapPut(UpdateSheet, "{id}")
+            .MapDelete(DeleteSheet, "{id}");
+    }
+
+    public Task<PaginatedList<SheetAdminListItemVm>> GetSheets(ISender sender, [AsParameters] GetSheetsWithPaginationQuery query)
+        => sender.Send(query);
+
+    public Task<List<SheetUserListItemVm>> GetUserSheets(ISender sender) => sender.Send(new GetSheetsByUserIdQuery());
+
+    public Task<SheetVm> GetSheet(ISender sender, int id) => sender.Send(new GetSheetByIdQuery(id));
```

Comment:
Shouldn't these return Action Result?

Reply:
I can't use Action Result because in this template the controllers (Endpoints) don't inherit from ControllerBase. I can use IResult to specify the return as you can see in the image but I have tested it and the behaviour of using IResult and not using it is kind of the same.

![image](https://github.com/zetTtai/DnDCharacterSheet/assets/87496967/12d7fd3a-237c-436a-b035-f067bbdc54cb)